src/core.rs

```diff
@@ -64,8 +64,13 @@ impl Core {
 
     fn fetch(&self, paths: Vec<PathBuf>) -> Vec<Meta> {
         let mut meta_list = Vec::with_capacity(paths.len());
+        let tree_layout_true = Layout::Tree { long: true };
+        let tree_layout_false = Layout::Tree { long: false };
 
-        let depth = if self.flags.recursive || self.flags.layout == Layout::Tree {
+        let depth = if self.flags.recursive
```

Comment:
I guess we could switch it out to this.

```rust
        let depth = match &self.flags.layout {
            &Layout::Tree { long } if long || !long || self.flags.recursive => {
                self.flags.recursion_depth
            }
            _ => 1,
        };
```

Reply:
Thx, I was looking for a better solution but couldn't get it ahah.